File: Segundo_Curso/Segundo_Cuatri/PDOO/ruby/Irrgarten/GameState.rb
#encoding: UTF-8
module GameState

    # Clase encargada de la distribución del area de juego
    # @param labyrinthv Estado del laberinto
    # @param players Estado de los jugadores
    # @param monsters Estado de los monstruos
    # @param current_players Indice del jugador actual
    # @param winner Indicador de si hay un vencedor
    # @param log Almacen de eventos del turno anterior
    class GameState
		
		# Inicializador de la clase
		# @param laby Estado del laberinto
		# @param avatars Estado de los jugadores
		# @param beasts Estado de los monstruos
		# @param p_player Indice del jugador actual
		# @param victorious Indicador de vencedor
		# @param account Almacen de eventos del turno anterior
		# @pos La instacia queda inicializada al completo
		def initialize(laby,avatars,beasts,p_player,victorious,account)
			@labyrinth = laby
			@players = avatars
			@monsters = beasts
			@current_player = p_player
			@winner = victorious
			@log = account
		end
		
		# Consultor del estado del laberinto
		# return Estado del laberinto

		def labyrinth
			@labyrinth
		end

		# Consultor del estado de los jugadores
		# return Estado de los jugadores			
		def players
			@players
		end

		# Consultor del estado de los monstruos
		# return Estado de los monstruos
		def monsters
			@monsters
		end
		
		# Consultor del indice del jugador actual
		# return Indice del jugador actual
		def current_player
			@current_player
		end
		
		# Consultor del fin de la partida por vencedor
		# return true si hay vencedor
		# 	 false si no lo hay
		def winner
			@winner
		end
		
		# Consultor del almacen de eventos del turno anterior
		# return Eventos del turno anterior
		def log
			@log
		end
    end

# Main

estadio = GameState.new('Carpena','juan','varios',5,true,'dramas.es')

puts estadio.inspect

puts estadio.labyrinth

puts estadio.players

puts estadio.monsters

puts estadio.current_player

puts estadio.winner

puts estadio.log

end

File: Segundo_Curso/Segundo_Cuatri/PDOO/java/src/irrgarten/GameState.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author el_dramas
 */
public class GameState {

    /**
     *  Estado del laberinto 
     */
    private String labyrinth;
    /**
     *  Estado de los jugadores
     */
    private String players;
    /**
     *  Estado de los montruos
     */
    private String monsters;
    /**
     *  Índice del jugador al que le toca jugar
     */
    private int currentPlayer;
    /**
     *  Estado de la partida, si hay un ganador se establece el valor true.
     */
    private boolean winner;
    /**
     *  Log de la partida, recoge todo lo ocurrido desde el último turno
     */
    private String log;
    
    /**
     * Constructor: Construye un GameState a partir de los siguientes parámetros
     * @param laby El laberinto
     * @param avatars Los jugadores
     * @param beasts Los monstruos
     * @param principalPlayer El primer jugador
     * @param victorious El ganador
     * @param _log El log inicial 
     */
    GameState(String laby, String avatars, String beasts,int principalPlayer, boolean victorious, String _log){
        labyrinth = laby;
        players = avatars;
        monsters = beasts;
        currentPlayer = principalPlayer;
        winner = victorious;
        log = _log;
        
    }
    
    /** 
     * Devuelve el estado del laberinto actual
     * @return El estado del laberinto actual
     */
    public String getLabyrinthv(){
        return labyrinth;
    }
    
    /**
     * Devuelve los jugadores actuales
     * @return Los jugadores actuales
     */
    public String getPlayers(){
        return players;
    }
    
    /**
     * Devuelve los monstruos actuales
     * @return Los monstruos actuales
     */
    public String getMonsters(){
        return monsters;
    }
    
    /**
     * Devuelve el jugador al que le toca
     * @return El índice jugador al que le toca
     */
    public int getCurrentPlayer(){
        return currentPlayer;
    }
    
    /**
     * Devuelve si hay un ganador
     * @return True si hay un ganador
     */
    public boolean getWinner(){
        return winner;
    }
    
    /**
     * Devuelve el log del último turno
     * @return El log del último turno
     */
    public String getLog(){
        return log;
    }
    
}  
